pykern/pkio.py

```diff
@@ -285,14 +271,17 @@ def save_chdir(dirname, mkdir=False, is_pkunit_prefix=False):
             pkunit_prefix = prev_ppp
 
 
-def sorted_glob(path, key=None):
-    """sorted list of py.path.Local objects
+def sorted_glob(pattern, key=None):
+    """Returns sorted list of files & dirs matching pattern.
 
-    Use '**' in path for a recursive search
+    Use '**' in pattern for recursive search, else, use * as wildcard.
+    Sorts using key if provided, else in ascending order.
+    Doesn't include hidden files unless . in pattern.
```

Comment:
unless dot "." is included explicitly at the start of a path component. 

BTW, If you do '.*' it only includes dot files. The same is true of '.**'. '**' does not include dot files even though it is recursive. Might be good to mention usage: sorted_glob is not usually recursive, and for recursive, it's probably better to use walk_tree and a file_re.

Reply:
sorted_glob is fine for what it was intended for: pattern match some files. Adding the recurse=True was a mistake.

My impression is that `**` is expensive, but really depends on the implementation. It was [faster in 2012](https://stackoverflow.com/a/27565420) than os.walk, but then read the last comment about how 3.5 fixed the performance. 

These type of performance differences matter less than functionality. sorted_glob was, again, for a quick glob match that was sorted (Py glob doesn't sort), e.g. find all *.log files in order time order. There are uses of '**' mostly in tests. The one case in Sirepo is files_to_watch_for_reload probably would be better off with walk_tree, since the query is being executed NxM when it could be executed once with a combined regex for the paths to match. 

In the end, it's like a lot of things: there's more than one way, and we have personal preferences or what we can remember to pick one or the other. Programmers don't often agree on these things. :)

BTW, glob include_hidden was added in 3.11, which is false by default.